Simplify main.jsx

b-tess opened the pull request:
## Description

Please include a summary of the changes and the related issue.
Add any screenshots if a change has been made on the client side.

Move routes to App.jsx

Move products page to pages folder

Manage products state in features/products folder

Add a home route/page

<img width="960" alt="Screenshot 2024-07-24 234258" src="https://github.com/user-attachments/assets/0b2cb3fd-62d6-4003-8fae-7033cd0d00cb">


## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

b-tess committed:
Simplify main.jsx

Move routes to App.jsx

Move products page to pages folder

Manage products state in features/products folder

Add a home route/page

SFrasia reviewed (approved):
Lookd good

b-tess committed:
Include serialized relationships between products and stores tables

Add some columns to the products table

Add supply request table

Include a serialized one-to-many relationship between products
and supply_requests

Add routes to get and post supply requests

Edit seed.py to include relevant data for the stores, products,
and supply_requests tables

otsembo commented on client/vite.config.js:
```diff
@@ -3,5 +3,15 @@ import react from '@vitejs/plugin-react'
+        cors: true,
+        proxy: {
+            '/api': {
+                target: 'http://127.0.0.1:5000',
```
You may need to add the target URL as an environment variable so that when the app is run in production it uses the correct URL in the different environments.

b-tess replied:
Hi @otsembo . Isn't the server property ignored in a production environment when it comes to Vite? I had come across that somewhere but I could be wrong. I'll move it to an environment variable but I'm curious about whether this property counts in production.

otsembo replied:
That is actually true. You are correct, it will be ignored in a production environment.

otsembo reviewed (changes requested):
@b-tess Check comment for minor change.

b-tess committed:
Move the database URL to an environment variable

Use this environment variable in vite.config.js

Start on displaying the product items from the API using
relevant data from updated backend

Explain why I commented out some parts of the models
on the backend. These comments can be removed later
once potential merge conflicts are resolved

Seed the tables with additional info relevant to the frontend

b-tess committed:
Add sqlalchemy-serializer to Pipfile

b-tess committed:
Slight change in the layout styling to make it more usable

Style products display on the frontend

Add formik and Yup for form data validation

Add some default style features to tailwind.config.js

Put the supply requests in their own feature for better
usability, flow and understanding

Start on supply request functionality

otsembo reviewed (approved):
Looks good.
